common/namespace/nsregistry/registry.go

```diff
@@ -114,9 +110,11 @@ type (
 		refreshInterval         dynamicconfig.DurationPropertyFn
 
 		// cacheLock protects cachNameToID, cacheByID and stateChangeCallbacks.
-		cacheLock                     sync.RWMutex
-		cacheNameToID                 cache.Cache
-		cacheByID                     cache.Cache
+
+		registryLock  sync.RWMutex
+		nameToID      map[namespace.Name]namespace.ID
+		idToNamespace map[namespace.ID]*namespace.Namespace
```

Comment:
It was clear before that these 3 fields play together. While I agree, `cache` was a bad word for grouping them, not having it at all is not good either. What's about:
```suggestion
		mapsLock  sync.RWMutex
		nameToIDMap      map[namespace.Name]namespace.ID
		idToNamespaceMap map[namespace.ID]*namespace.Namespace
```
?

Reply:
Ideally, it would be great to create another struct which combines these 3 fields.

Reply:
you added "map" to those fields.
*  xxToYy already imply by convention its a map
* adding type to a field name is generally considered a bad practice in modern programming

That said - yeah, registryLock is a bit generic, I will rename it to nsMapsLock